pkg/operator/deploy/staticresources/master/deployment.yaml.tmpl

```diff
@@ -79,4 +90,14 @@ spec:
         operator: Exists
       - effect: NoSchedule
         operator: Exists
-
+      {{ if .UsesWorkloadIdentity }}
+      volumes:
+      - name: bound-sa-token
+        projected:
+          defaultMode: 420
+          sources:
+          - serviceAccountToken:
+              audience: openshift
```

Comment:
This is just a question, but is this required?
Apparently this field is optional, so I'm curious why we need it.

Reply:
My understanding is that the value of the `audience` field needs to match what's in the token itself, otherwise it will be rejected

![image](https://github.com/user-attachments/assets/01683521-3ded-4bda-a8b6-e00fb98eae9c)

@cadenmarchese is this accurate? Do we need the `audience` field?

Reply:
According to @bennerv it is required. Linking the thread with explanation: https://redhat-external.slack.com/archives/C06FRQKUMFF/p1723242231734469

> OpenShift hardcodes all it's operators & deployments to mount a service account token for the audience: openshift instead of using azure.workload.identity/use: true label on the pod spec.
This is because the cluster version operator doesn't know (or doesn't care) if you're using workload identity when it lays down the manifests for the operators.
So the audience on the federated credentials for these identities is openshift.  if you end up using the azure.workload.identity/use label on the ARO operator for workload identity enabled clusters, you'll have a different audience in the federated credentials.  It'll be api://AzureADTokenExchange